### PR TITLE
new: added Cmd#parse_with method.

### PR DESCRIPTION
**Pull request by sttk**
This PR adds `parse_with` method to `Cmd` struct.
This method is to parse command line argument with option configurations which are held in `OptCfg` structs.

### CODE REVIEW

File: src/parse/parse_with.rs [hunk 1]
```diff
@@ -0,0 +1,1474 @@
...
+
+        assert_eq!(cmd.name(), "app");
+        assert_eq!(cmd.has_opt("foo"), false);
+        assert_eq!(cmd.has_opt("bar"), false);
+        assert_eq!(cmd.has_opt("f"), false);
+        assert_eq!(cmd.args(), &[] as &[&str]);
+    }
+
+    #[test]
+    fn use_store_key() {
```
Comment (sttk) [inline src/parse/parse_with.rs:1360]: Add a test case that use `store_key` as option name if `names` is empty.

**Review by sttk — COMMENTED** [review]
There are a few points, so take care of them.
In addtion, add tests in `tests` directory.
